internal/beef/beef.go

```diff
@@ -0,0 +1,277 @@
+package beef
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/libsv/go-bt/v2"
+)
+
+const (
+	versionBytesCount = 2
+	markerBytesCount  = 2
+	hashBytesCount    = 32
+	maxTreeHeight     = 64
+)
+
+const (
+	BEEFMarkerPart1 = 0xBE
+	BEEFMarkerPart2 = 0xEF
+)
+
+const (
+	HasNoBump = 0x00
+	HasBump   = 0x01
+)
+
+type TxData struct {
+	Transaction *bt.Tx
+	BumpIndex   *bt.VarInt
+
+	txID string
+}
+
+type BEEF struct {
+	BUMPs        BUMPs
+	Transactions []*TxData
+}
+
+func CheckBeefFormat(txHex []byte) bool {
+	if len(txHex) < versionBytesCount+markerBytesCount {
```

Comment:
```suggestion
	if len(txHex) < 4 {
```
It's known from the BRC

Reply:
Yes, but I hate magic numbers. If there's a random number in the code, it should be a named-variable so you know what it means. 

But I refactored it and removed these two variables in favour of one.